frontend/test/resolution/testSplitInit.cpp

```diff
@@ -0,0 +1,860 @@
+
+#include "chpl/parsing/parsing-queries.h"
+#include "chpl/resolution/resolution-queries.h"
+#include "chpl/resolution/scope-queries.h"
+#include "chpl/uast/Call.h"
+#include "chpl/uast/Comment.h"
+#include "chpl/uast/Identifier.h"
+#include "chpl/uast/Module.h"
+#include "chpl/uast/Variable.h"
+
+#include "./ErrorGuard.h"
+
+static const bool ERRORS_EXPECTED=true;
+
+// resolves the last function
+// checks that the set of split init variables matches
+// the array of variable names
+static void testSplitInit(const char* test,
+                          const char* program,
+                          std::vector<const char*> expectedSplitInits,
+                          bool expectErrors=false) {
+  printf("%s\n", test);
+
+  Context ctx;
+  Context* context = &ctx;
+  ErrorGuard guard(context);
+
+  std::string testname = test;
+  testname += ".chpl";
+  auto path = UniqueString::get(context, testname);
+  std::string contents = program;
+  setFileText(context, path, contents);
+
+  const ModuleVec& vec = parseToplevel(context, path);
+  assert(vec.size() == 1);
+  const Module* M = vec[0]->toModule();
+  assert(M);
+  assert(M->numStmts() >= 1);
+
+  const Function* func = M->stmt(M->numStmts()-1)->toFunction();
+  assert(func);
+
+  // resolve runM1
+  const ResolvedFunction* r = resolveConcreteFunction(context, func->id());
+  assert(r);
+
+  std::set<ID> splitIds = computeSplitInits(context, func, r->resolutionById());
+  std::set<std::string> splitNames;
+  for (auto id : splitIds) {
+    auto ast = idToAst(context, id);
+    auto vl = ast->toVarLikeDecl();
+    splitNames.insert(vl->name().str());
+  }
+
+  std::set<std::string> expectNames;
+  for (auto cstr : expectedSplitInits) {
+    expectNames.insert(std::string(cstr));
+  }
+
+  // check that each thing in expectNames is in splitNames
+  for (auto expectName : expectNames) {
+    if (splitNames.count(expectName) == 0) {
+      printf("%s: missing expected split init for '%s'\n",
+             test, expectName.c_str());
+    }
+  }
+
+  // check that each thing in splitNames is in expectNames
+  for (auto splitName : splitNames) {
+    if (expectNames.count(splitName) == 0) {
+      printf("%s: unexpected split init for '%s'\n", test, splitName.c_str());
+    }
+  }
+
+  assert(expectNames == splitNames);
+
+  // check that all split-inited variables have type 'int'
+  auto intType = IntType::get(context, 0);
+  for (auto id : splitIds) {
+    QualifiedType varType = r->byId(id).type();
+    if (varType.type() != intType) {
+      auto ast = idToAst(context, id);
+      auto vl = ast->toVarLikeDecl();
+      auto name = vl->name().str();
+      printf("%s: variable '%s' with id %s has unexpected type ",
+             test, name.c_str(), id.str().c_str());
+      varType.dump();
+      printf(" rather than type 'int'\n");
+    }
+    assert(varType.type() == IntType::get(context, 0));
+  }
+
+  size_t errCount = guard.errors().size();
+  guard.realizeErrors();
```

Comment:
I thought this can be `int errCount = guard.realizeErrors()`, since `realizeErrors` returns the number of errors.